src/server/modules/threat/services/FileTypeFilterService.ts

```diff
@@ -50,12 +50,23 @@ export class FileTypeFilterService implements interfaces.FileTypeFilterService {
     this.allowedFileExtensions = allowedFileExtensions
   }
 
-  hasAllowedType: (file: { name: string; data: Buffer }) => Promise<boolean> =
-    async ({ name, data }) => {
-      const fileType = await FileType.fromBuffer(data)
-      const extension = fileType?.ext || `${`${name}`.split('.').pop()}`
-      return this.allowedFileExtensions.includes(extension)
+  hasAllowedType: (
+    file: {
+      name: string
+      data: Buffer
+    },
+    inputExtension?: string,
```

Comment:
Nit/ naming suggestion, can we rename this to `allowedType(s)/ allowedExtensions` so that it's consistent with the function name? 

Also a suggestion, maybe you can generalise this too so that somebody can pass in an array of allowed input extensions, instead of just a single one.